test/Parse/enum_element_pattern_swift4.swift

```diff
@@ -9,10 +9,11 @@ enum E {
 
   static func testE(e: E) {
     switch e {
-    case A<UndefinedTy>(): // expected-error {{cannot specialize a non-generic definition}}
+    case A<UndefinedTy>(): // expected-error {{cannot find type 'UndefinedTy' in scope}}
```

Comment:
Maybe this is something that could be addressed during pre-check which would set a placeholder type for `UndefinedTy` declaration in this context?

Reply:
Yep, fixed!